src/parser/ir/Ir.g4

```diff
@@ -57,10 +58,21 @@ ANY: '<<ANY>>';
 BLOCK: 'block';
 MODULE: 'module';
 
-
+SIGN : '+' | '-';
 NUMLITERAL : [0-9]+;
+FRACTIONAL_PART : '.' [0-9]*;
+EXPONENT: ('e'|'E') SIGN? [0-9]+;
+DOUBLE_TAIL
+    : (FRACTIONAL_PART EXPONENT? ('l' | 'f')?)
```

Comment:
Does it make sense to have a distinction between `l` and `f`? If we put everything in double precision, we don't really need `f`.

Reply:
I've dropped support for 'f'. I had introduced it to attempt to match C++ (with the intention of adding float/double values, but I think the higher accuracy values are better for our purposes anyway).